Mathlib/Data/List/Infix.lean

```diff
@@ -70,6 +70,13 @@ theorem infix_rfl : l <:+: l :=
 theorem prefix_concat (a : α) (l) : l <+: concat l a := by simp
 #align list.prefix_concat List.prefix_concat
 
+theorem prefix_concat_iff {x y : List α} {a : α} : x <+: y ++ [a] ↔ x = y ++ [a] ∨ x <+: y := by
+  convert suffix_cons_iff using 1
+  · rw [← List.reverse_concat, List.reverse_suffix, ← concat_eq_append]
+  · congr! 1
+    · rw [← List.reverse_concat, ← concat_eq_append, reverse_inj]
+    · apply reverse_suffix.symm
```

Comment:
Could you get a proof looking like this working?
```suggestion
  simpa using suffix_cons_iff (l₁ := x.reverse) (l₂ := y.reverse)
```
(Also this shows you probably shouldn't call your lists `x` and `y`)

Reply:
Okay, this worked after adding a new lemma reverse_concat' (which seems to be good to have anyway as `x++[a]` seemingly is the simp-normal form of `x.concat a`. About the names: Is there a consistent convention? I think the mathlib naming convention mentioned s,t,... for lists, parts of the library use l_i, etc.

Reply:
`s` , `t` for lists is quite strange. I use `a b c d w x y z : α`, `l l₁ l₂ : List α`, `L L₁ L₂ : List (List α)`, `m m₁ m₂ s s₁ s₂ t t₁ t₂ : Multiset α`, `s s₁ s₂ t t₁ t₂ u v : Finset α`.

Reply:
Maybe in this case https://leanprover-community.github.io/contribute/style.html should be updated

Reply:
Yes, feel free to open a PR for that